services/src/handlers/datasets.rs

```diff
@@ -869,6 +873,38 @@ fn column_map_to_column_vecs(columns: &HashMap<String, ColumnDataType>) -> Colum
     }
 }
 
+/// Delete a dataset
+#[utoipa::path(
+    tag = "Datasets",
+    delete,
+    path = "/dataset/{dataset}",
+    responses(
+        (status = 200, description = "OK"),
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+pub async fn delete_dataset_handler<C: Context>(
+    dataset: web::Path<DatasetId>,
+    session: AdminOrSession<C>,
+    ctx: web::Data<C>,
+) -> Result<impl Responder>
+where
+    C::Session: MockableSession,
+{
+    let session = match session {
+        AdminOrSession::Admin => AdminSession::default().into(),
+        AdminOrSession::Session(s) => s,
+    };
+
+    ctx.dataset_db_ref()
+        .delete_dataset(&session, dataset.into_inner())
+        .await?;
+
+    Ok(actix_web::HttpResponse::Ok())
```

Comment:
If we don't return anything, should we send a 204 or something?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200 states

> The successful result of a [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) or a [DELETE](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE) is often not a 200 OK but a [204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) No Content (or a [201](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) Created when the resource is uploaded for the first time).

Reply:
good point. We could also return an empty string, or "success" or something like that. 200 and no content is not ideal I guess.

Reply:
developer.mozilla suggest 204 (cf. above), so why is it bad?

Reply:
We can do it this way, but then we can no longer rely on status code = 200 on success. Status code = 200 => success is also very handy. Especially if we decide at a later point in time to do return something after all. Then we don't have to change the client code.

But: I am not against it, it's just not enough of a no-brainer for me to change it right away.